chia/wallet/nft_wallet/uncurry_nft.py

```diff
@@ -166,7 +168,7 @@ def uncurry(cls: Type[_T_UncurriedNFT], mod: Program, curried_args: Program) ->
             return None
 
         return cls(
-            nft_mod_hash=nft_mod_hash,
+            nft_mod_hash=bytes32.from_bytes(bytes(nft_mod_hash)[1:]),
```

Comment:
You don't want to turn Program objects into bytes32 like this even though it will work.  The `a0` length prefix that you're stripping off here would be different for other, longer blobs.  You should instead do `bytes32(nft_mod_hash.atom)`.